### ISSUE TITLE
spark-operator examples are using invalid image (gcr.io) and switching to ghcr.io breaks them anyway

### ISSUE THREAD

**Issue by PuszekSE** [kubeflow/spark-operator#1888]
I've been trying to setup the operator, checking it with the examples provided in the repo:
https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/examples/spark-pi.yaml

Obviously I had to modify that yaml as it's using old, gcr.io docker image link, so the only changes applied to the file were:
*  image: "ghcr.io/googlecloudplatform/spark-operator:v1beta2-1.3.2-3.1.1"
*  namespace: [spark-operator-namespace]

In the end the 'spark-pi-driver' gets created, by the process/logs stop on:
```+ exec /usr/bin/tini -s -- /usr/bin/spark-operator driver --properties-file /opt/spark/conf/spark.properties --class org.apache.spark.examples.SparkPi local:///opt/spark/examples/jars/spark-examples_2.12-3.1.1.jar ```
The executors never get spawned nothing in the logs, Spark UI is not started and the driver pod just keeps running forever.

I've logged into the pod. Running the SparkPi example manually via `/opt/spark/bin/run-example` does work and spark.properties do seem to have valid contents.
I've tried running `/usr/bin/spark-operator` manually (it's what entrypoint uses), but it hangs as well with no console output, no matter whether the input was valid or not - (ie. I've tried running it with non-existing main class and the effect was the same).

I have no idea whether I've installed the operator wrong, the ghcr.io image is invalid or if there's something completely different wrong going on.

**Comment by PuszekSE** [comment]
To follow up on that, I've decided to check that using fresh minikube cluster to make sure it didn't have anything to do with wrong configuration on that side.

What I had to add manually were service account and cluster roles to even deploy the driver, as funnily enough [configuration](https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/charts/spark-operator-chart/README.md#values) for that reported:
```Error: unknown flag: --serviceAccounts.spark.create```

With some extremely basic configs (below) I've managed to deploy the driver, as I did initially, but the results were exactly the same - spark-operator process was started on the driver and nothing else has happened... It makes me wonder if somehow it has to do with a different docker image (ghcr.io vs gcr.io) although that's really a stretch...

```rules:
  - apiGroups: ["*"]
    resources: ["*"]
    verbs: ["*"]```

**Comment by reminia** [comment]
@PuszekSE  I guess you're using a wrong image like `ghcr.io/googlecloudplatform/spark-operator:v1beta2-1.3.3-3.1.1`, this image is for spark-operator. You should use a spark image like `apache/spark:3.3.1`.

**Comment by PuszekSE** [comment]
That's a bit problematic, considering that examples in the repository, for running spark applications via operator are actually using this specific image. (or actually it's not-operational mirror from gcr.io)

I'll edit the issue accordingly

**Comment by reminia** [comment]
@PuszekSE  yep, the image u used is for spark-operator not for spark. That's the problem.  
It's released regularly for the operator.Try what I said, using a spark image instead.
I tried the same image as you before. It took me a lot of time to figure it out.
Looks like the project has not been updated for months and no one is active to solve the issues. :)

**Comment by PuszekSE** [comment]
I've already made it work with the default [apache-spark](https://hub.docker.com/r/apache/spark/tags) image (specifically v3.1.3).

I believe it's still valid to leave this issue itself open, even for visibility and for others to know how to avoid it until the repositories will get cleaned up.

**Comment by Suozz** [comment]
> I've already made it work with the default [apache-spark](https://hub.docker.com/r/apache/spark/tags) image (specifically v3.1.3).
> 
> I believe it's still valid to leave this issue itself open, even for visibility and for others to know how to avoid it until the repositories will get cleaned up.

24/01/10 13:39:38 WARN DependencyUtils: Local jar /opt/spark/examples/jars/spark-examples_2.12-3.1.1.jar does not exist, skipping.
24/01/10 13:39:38 WARN DependencyUtils: Local jar /opt/spark/examples/jars/spark-examples_2.12-3.1.1.jar does not exist, skipping.
Error: Failed to load class org.apache.spark.examples.SparkPi.

I met this problem using [apache-spark](https://hub.docker.com/r/apache/spark/tags):v3.1.3 

How to fix this?

**Comment by reminia** [comment]
@Suozz  Your spark-pi job is using 3.1.1 from the WARN log, you can fix it by using apache-spark v3.1.1.
